examples/gno.land/p/demo/auction/auction.gno

```diff
@@ -0,0 +1,140 @@
+package auction
+
+import (
+	"std"
+	"strconv"
+	"time"
+
+	"gno.land/p/demo/avl"
+)
+
+// Main struct
+type Auction struct {
+	title       string
+	owner       std.Address
+	description string
+	begin       time.Time
+	end         time.Time
+	price       uint64
+	bids        *avl.Tree // key: std.Address, value: Bid
+}
+
+type Bid struct {
+	bidder std.Address
+	amount uint64
+}
+
+type EventEnd struct {
+	winner std.Address
+	amount uint64
+}
+
+func NewAuction(
+	title string,
+	owner std.Address,
+	description string,
+	begin time.Time,
+	end time.Time,
+	minPrice uint64,
```

Comment:
What do you think about having this `minPrice` renamed to `startingBid`?
And it being an `std.Coin`?

Reply:
Thanks for the suggestion! I've renamed` minPrice` to `startingBid`. I've also changed its type to std.Coin, which makes it possible to handle different currencies and facilitates comparisons and calculations related to auction amounts [af0ae2e1](https://github.com/gnolang/gno/pull/2265/commits/af0ae2e1319fa998ae85ee2ce7587edcef87daf4)